numpy/_core/tests/_natype.py

```diff
@@ -38,11 +38,11 @@ def method(self, other):
             return pd_NA
 
         elif isinstance(other, np.datetime64):
-            if name in ["__sub__", "__rsub__"]:
+            if name in {"__sub__", "__rsub__"}:
                 return pd_NA
 
         elif isinstance(other, np.timedelta64):
-            if name in ["__sub__", "__rsub__", "__add__", "__radd__"]:
+            if name in {"__sub__", "__rsub__", "__add__", "__radd__"}:
                 return pd_NA
```

Comment:
probably out of scope, but these could be combined to avoid duplication of the `__r?sub__` code